include/openthread/dataset.h

```diff
@@ -213,6 +213,17 @@ typedef struct otOperationalDatasetComponents
     bool mIsChannelMaskPresent : 1;      ///< TRUE if Channel Mask is present, FALSE otherwise.
 } otOperationalDatasetComponents;
 
+/**
+ * This struct represents a Thread Dataset timestamp.
+ *
+ */
+typedef struct otTimestamp
+{
+    uint64_t mSeconds : 48;
+    uint64_t mTicks : 15;
+    uint64_t mAuthoritative : 1;
```

Comment:
```suggestion
    union {
      struct {
        uint64_t mSeconds : 48;
        uint64_t mTicks : 15;
        uint64_t mAuthoritative : 1;
      } mFields;
      uint64_t mTimestamp;
    }
```

May use a union to allow accessing the full uint64_t easily?

Reply:
It simply won't work as C/C++ doesn't specify the order of bit fields, compiler implementation can use different orders.  See the `Notes` section in https://en.cppreference.com/w/cpp/language/bit_field.

Reply:
Thanks for the note. 

SGTM, even though I prefer an `uint64_t` underlying representation slightly.